frame/contracts/COMPLEXITY.md

```diff
@@ -187,6 +188,20 @@ implementation they just involve a DB read.
 For subsequent calls and instantiations during contract execution, the initialization requires no
 expensive operations.
 
+## Terminate
+
+This function performs the following steps:
+
+1. Check the calling contract is not already on the callstack by calling `is_live`.
+2. `transfer` funds from caller to the beneficiary.
+3. Flag the caller contract as deleted in the overlay.
+
+`is_live` does not do any database access nor does it allocate memory. It walks up the call
+stack and therefore executes in linear time depending on size of the call stack.
```

Comment:
Which is also has upper bound (IIRC, 32 in size) so can be considered constant.

Reply:
Added that information.